app/src/main/java/com/arctouch/codechallenge/data/remote/MovieRemoteDataSource.java

```diff
@@ -0,0 +1,46 @@
+package com.arctouch.codechallenge.data.remote;
+
+import androidx.annotation.NonNull;
+
+import com.arctouch.codechallenge.data.MoviesDataSource;
+import com.arctouch.codechallenge.data.model.GenreResponse;
+import com.arctouch.codechallenge.data.model.Movie;
+import com.arctouch.codechallenge.network.MovieRemoteApi;
+
+import java.util.List;
+
+import io.reactivex.Observable;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.schedulers.Schedulers;
+
+public class MovieRemoteDataSource extends MoviesDataSource {
+
+    private MovieRemoteApi movieApi;
+
+    public MovieRemoteDataSource(){
+        movieApi = MovieRemoteApi.getInstance();
+    }
+
+    @Override
+    public Observable<List<Movie>> subscribeForAllmovies(String language, int page, String region) {
```

Comment:
What would be your strategy to unit test this method? Can you foresee any difficulties you might face?

Reply:
I would try to use parameters that I know that a small number of objects would come, and what informations those objects were storing. Then would try to compare with a mock object and assert true if equals, assert not null. The deficcults would be to emulate this request e unit test an observable is also a challenge.